pkg/controller/replication/replication_controller.go

```diff
@@ -408,21 +428,26 @@ func (r *ReconcileReplication) getHarborRegistryFromRef(ctx context.Context, reg
 		return nil, internal.ErrRegistryNotReady(registry.Name)
 	}
 
-	return registry.Spec.ToHarborRegistry(), nil
+	var registryID int64
+
+	if registry.Status.ID != 0 {
+		registryID = registry.Status.ID
+	}
+
+	return registry.Spec.ToHarborRegistry(registryID), nil
```

Comment:
Why you are using the extra `registryID` var?

```suggestion
	return registry.Spec.ToHarborRegistry(registry.Status.ID), nil
```